blog/views.py

```diff
@@ -0,0 +1,70 @@
+from django.views.generic import ListView, DetailView, \
+   CreateView, UpdateView, DeleteView
+from .models import Post, Category
+from .forms import PostForm, EditForm
+from django.urls import reverse_lazy
+from django.shortcuts import render
+from django.views import generic
+from django.contrib.auth.forms import UserCreationForm
+
+
+class HomeView(ListView):
+    model = Post
+    template_name = 'home.html'
+    ordering = ['-post_date']
+
+    def get_context_data(self, *args, **kwargs):
+        cat_menu = Category.objects.all()
+        context = super(HomeView, self).get_context_data()
+        context["cat_menu"] = cat_menu
+        return context
+
+
+def CategoryListView(request):
+    cat_menu_list = Category.objects.all()
+    return render(request, 'category_list.html', {
+       'cat_menu_list': cat_menu_list
+        })
+
+
+def CategoryView(request, cats):
```

Comment:
Move logic with filtration by category to separate PR and use Based Views

Reply:
done